<launch>
    <node pkg="duo3d_driver" type="duo3d_node" name="collect_stereo"/>

    <node pkg="image_rotate" type="image_rotate" name="flip_l">
	<remap from="image" to="/stereo/left/image_rect"/>
	<remap from="rotated/image" to="/stereo/rotate/left/image_rect"/>
    </node>

    <node pkg="image_rotate" type="image_rotate" name="flip_r">
	<remap from="image" to="/stereo/right/image_rect"/>
	<remap from="rotated/image" to="/stereo/rotate/right/image_rect"/>
    </node>

    <node pkg="sparse_stereo" type="sparse_stereo_opencv_matching" name="sparse_stereo" output="screen">
	<remap from="/kitti/camera_gray_left/image_raw" to="/stereo/rotate/left/image_rect"/>
        <remap from="/kitti/camera_gray_left/camera_info" to="/stereo/left/camera_info"/>
	<remap from="/kitti/camera_gray_right/image_raw" to="/stereo/rotate/right/image_rect"/>
        <remap from="/kitti/camera_gray_right/camera_info" to="/stereo/right/camera_info"/>
    </node>

    <node pkg="sparse_stereo" type="local_octomap_generator" name="octomap">
	<remap from="/points2" to="/sparse_points2"/>
    </node>

    <!--node pkg="baseline_testing" type="octomap_controller_node" name="navigation">
	<remap from="camera/depth/points" to="/sparse_points2"/>
    </node-->

    <node pkg="nodelet" type="nodelet" name="pointcloud" args="standalone stereo_image_proc/point_cloud2">
	<remap from="left/image_rect_color" to="/stereo/left/image_rect"/>
	<remap from="left/camera_info" to="/stereo/left/camera_info"/>
	<remap from="right/camera_info" to="/stereo/right/camera_info"/>
	<remap from="disparity" to="/stereo/disparity"/>

	<param name="approximate_sync" type="bool" value="true"/>
	<param name="queue_size" type="int" value="12"/>
    </node>

    <node pkg="image_view" type="disparity_view" name="show_disparity" args="image:=/stereo/disparity"/>

    <node pkg="image_view" type="image_view" name="show_left_image" args="image:=/stereo/left/image_rect"/>
    <node pkg="image_view" type="image_view" name="show_right_image" args="image:=/stereo/right/image_rect"/>

    <node pkg="rviz" type="rviz" name="show_pc" args="-d $(find sparse_stereo)/rviz/sparse_stereo_gazebo.rviz"/>

</launch>
